src/main/java/org/apache/tomcat/jakartaee/MigrationCache.java

```diff
@@ -0,0 +1,498 @@
+                    continue;
+                }
+
+                String[] parts = line.split("\\|");
+                if (parts.length == 2) {
+                    String hash = parts[0];
+                    try {
+                        LocalDate lastAccessed = LocalDate.parse(parts[1], DATE_FORMATTER);
+                        cacheMetadata.put(hash, lastAccessed);
+                    } catch (DateTimeParseException e) {
+                        logger.log(Level.WARNING, sm.getString("cache.metadata.invalidDate", line));
+                    }
+                } else {
+                    logger.log(Level.WARNING, sm.getString("cache.metadata.invalidLine", line));
+                }
+            }
+
+            // Check for any cached files not in metadata and add them with today's date
+            Set<String> existingHashes = scanCacheDirectory(null);
+            for (String hash : existingHashes) {
+                if (!cacheMetadata.containsKey(hash)) {
+                    cacheMetadata.put(hash, today);
+                }
+            }
+
+            logger.log(Level.FINE, sm.getString("cache.metadata.loaded", cacheMetadata.size()));
+        } catch (IOException e) {
+            // Corrupt or unreadable - assume all cached files accessed today
+            logger.log(Level.WARNING, sm.getString("cache.metadata.loadError"), e);
+            cacheMetadata.clear();
+            scanCacheDirectory(today);
+        }
+    }
+
+    /**
+     * Scan cache directory for existing cache files and return their hashes.
+     * If accessDate is not null, adds all found hashes to metadata with that date.
+     *
+     * @param accessDate the date to use for all found files (null to not update metadata)
+     * @return set of hashes found in cache directory
+     */
+    private Set<String> scanCacheDirectory(LocalDate accessDate) {
+        Set<String> hashes = new HashSet<>();
+
+        File[] subdirs = cacheDir.listFiles();
+        if (subdirs != null) {
+            for (File subdir : subdirs) {
+                if (subdir.isDirectory()) {
+                    File[] files = subdir.listFiles();
+                    if (files != null) {
+                        for (File file : files) {
+                            if (file.isFile() && file.getName().endsWith(".jar")) {
+                                String hash = file.getName().substring(0, file.getName().length() - 4);
+                                hashes.add(hash);
+                                if (accessDate != null) {
+                                    cacheMetadata.put(hash, accessDate);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return hashes;
+    }
+
+    /**
+     * Check if caching is enabled.
+     *
+     * @return true if caching is enabled
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+
```

Comment:
This can also be removed. `cache != null` is sufficient. The `enabled` field can also be removed.